code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -396,6 +397,15 @@
 
 	return FALSE
 
+/mob/living/simple_animal/bot/medbot/proc/get_avoidchem_toxin(mob/M)
+	return toxlover_check(M)? null : treatment_tox_avoid
+
+/mob/living/simple_animal/bot/medbot/proc/get_healchem_toxin(mob/M)
+	return toxlover_check(M)? treatment_tox_toxlover : treatment_tox
+
+/mob/living/simple_animal/bot/medbot/proc/toxlover_check(mob/M)
+	return HAS_TRAIT(M, TRAIT_TOXINLOVER)

```

Comment:
This sort of functions that only return macros or simple operation without executing anything else should be avoided.

Reply:
aight bet